Switch to Gradle-Kotlin as the default build tool

timyates opened the pull request:
See: https://github.com/micronaut-projects/micronaut-starter/discussions/2062

timyates committed:
Switch to Gradle-Kotlin as the default build tool

timyates committed:
Fix tests expecting build.gradle

timyates committed:
Fix CLI defaults

timyates committed:
Fix Zip tests

sdelamo reviewed (approved):
I wonder if we have test coverage which tests everything we generate it is Gradle Kotlin DSL compatible.

timyates commented:
> I wonder if we have test coverage which tests everything we generate it is Gradle Kotlin DSL compatible.

Think it's worth spending some time looking for `BuildTool.GRADLE` only tests (with missing `BuildTool.GRADLE_KOTLIN`)?

timyates commented:
I'm going to mark this as pending whilst I work on the coverage in the test-feature modules

timyates committed:
Add gradle-kotlin to tests and fix Coherence

This commit adds gradle-kotlin tests to several in starter-core that were just testing the Groovy DSL.

It also fixes the Coherence feature where we were adding a version as a property.
This is problematic in the gradle-kotlin dsl as we would need to define it in the build script.
However I think in these days of Coherence having a BOM, it's also incorrect, and the version should come from there, not starter

timyates commented on starter-core/src/main/java/io/micronaut/starter/feature/coherence/CoherenceFeature.java:
```diff
@@ -32,11 +32,11 @@
 @Singleton
 public class CoherenceFeature implements Feature {
 
-    public static final String COHERENCE_VERSION = "21.06.2";
```
Previously we were adding this version to `gradle.properties`, and then using that as the version in the Gradle and Maven builds.

Now the Coherence module has a BOM, I believe this is the wrong behaviour (and it would require changes to the kotlin dsl scripts to import the property)

I've removed this from here, and the version of the `com.oracle.coherence.ce` artifacts instead comes from the coherence bom

timyates committed:
Fix cli config test

timyates commented:
Hmmm...rocker config needs fixing too I think...

timyates committed:
Fix Rocker config for kotlin and groovy, and extract common combinations for Gradle and languages

timyates commented on starter-core/src/main/java/io/micronaut/starter/feature/view/gradlePluginRocker.rocker.raw:
This works for both the Groovy and Kotlin DSL

timyates committed:
And then fix the test that tests the output is exact 🫠
